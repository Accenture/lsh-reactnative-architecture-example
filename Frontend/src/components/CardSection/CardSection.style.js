const cardSectionStyles = {
  containerStyle: {
    borderBottomWidth: 1,
    padding: 5,
    backgroundColor: '#fff',
    justifyContent: 'flex-start',
    flexDirection: 'row',
    borderColor: '#ddd',
    position: 'relative',
  },
};
export default cardSectionStyles;
