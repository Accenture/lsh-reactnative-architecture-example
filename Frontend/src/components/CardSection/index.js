import React from 'react';
import PropTypes from 'prop-types';
import { View } from 'react-native';
import styles from './CardSection.style';

const CardSection = props => (
  <View style={styles.containerStyle}>{props.children}</View>
);

CardSection.defaultProps = {
  children: PropTypes.element.isRequired,
};

CardSection.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node,
  ]),
};

export default CardSection;
