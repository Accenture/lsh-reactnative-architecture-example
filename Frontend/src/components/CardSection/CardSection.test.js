import React from 'react';
import { shallow } from 'enzyme';
import CardSection from './index';

describe('Tests card section component', () => {
  it('matches snapshot', () => {
    const comp = shallow(<CardSection />);
    expect(comp).toMatchSnapshot();
  });
});
