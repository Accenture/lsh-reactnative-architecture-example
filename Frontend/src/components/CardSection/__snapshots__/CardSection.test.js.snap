// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests card section component matches snapshot 1`] = `
<Component
  style={
    Object {
      "backgroundColor": "#fff",
      "borderBottomWidth": 1,
      "borderColor": "#ddd",
      "flexDirection": "row",
      "justifyContent": "flex-start",
      "padding": 5,
      "position": "relative",
    }
  }
/>
`;
